pkg/kubelet/dockershim/docker_service.go

```diff
@@ -309,6 +317,107 @@ type dockerService struct {
 
 // TODO: handle context.
 
+// GetRuntimeConfigInfo returns the runtime config.
+func (ds *dockerService) GetRuntimeConfigInfo(_ context.Context, r *runtimeapi.GetRuntimeConfigInfoRequest) (*runtimeapi.GetRuntimeConfigInfoResponse, error) {
+	dockerInfo, err := ds.client.Info()
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute Info() call to the Docker client: %v", err)
+	}
+	uidMapping := &runtimeapi.LinuxIDMapping{ContainerId: uint32(0)}
+	gidMapping := &runtimeapi.LinuxIDMapping{ContainerId: uint32(0)}
+
+	if isUserNsEnabled(dockerInfo) {
+		remappedNonRootHostID, err := getRemappedNonRootHostID(dockerInfo)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get remappedNonRootHostID. err: %v", err)
+		}
+		uidMappingSize, gidMappingSize, err := getUserNsMappingSizes(remappedNonRootHostID)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get user-namespace mapping sizes. err: %v", err)
+		}
+
+		uidMapping.ContainerId = uint32(0)
+		gidMapping.ContainerId = uint32(0)
+		uidMapping.HostId = remappedNonRootHostID
+		gidMapping.HostId = remappedNonRootHostID
+		uidMapping.Size_ = uidMappingSize
+		gidMapping.Size_ = gidMappingSize
```

Comment:
this seems to assume there can be only one mapping available for the user.  Is it safe to assume it?